modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMFormDisplayContext.java

```diff
@@ -796,7 +796,10 @@ public boolean isShowSubmitButton() {
 
 	public boolean isShowSuccessPage() throws PortalException {
 		if (!SessionErrors.isEmpty(_renderRequest) ||
-			SessionMessages.isEmpty(_renderRequest) ||
+			!SessionMessages.contains(
```

Comment:
Hey, @carolmariaabb. Just an idea... do you think we could accomplish the same result by checking the `_ddmFormInstanceRecordId`? I think Brian would notice that we're using the `SessionMessages` as a flags mechanism instead of, in fact, using it to display messages on the JSP. In case the `_ddmFormInstanceRecordId` doesn't work, do you think we could add this flag as a request parameter?

Reply:
I thought about using the `_ddmFormInstanceRecordId` but because the autosave there are cases where the `_ddmFormInstanceRecordId` is not 0 and we shouldn’t show de success page. I'll try to add as a request parameter.

Thanks!

Reply:
It makes sense, @carolmariaabb. If it's too much trouble and you think it looks uglier with the request parameter, feel free to discard the suggestion. It's better to simplify.

Reply:
I just remembered that I already tried to add this flag as a request parameter. But when I do `actionRequest.setAttribute("test", true)` in `AddFormInstanceRecordMVCActionCommand`, the ParamUtil.getBoolean(_renderRequest, "test") returns false  in `DDMFormDisplayContext`.
I even tried with the `portletSession.setAttribute` but (as expected 😝) after the first submission the success page is always shown.